pkg/util/cephcmds.go

```diff
@@ -182,6 +182,11 @@ func GetOMapValue(ctx context.Context, monitors string, cr *Credentials, poolNam
 			return "", ErrKeyNotFound{poolName + "/" + oMapName + "/" + oMapKey, err}
 		}
 
+		if strings.Contains(stdoutanderr, "error opening pool "+
```

Comment:
you need to return invalid request during the volume/snapshot Create operation. basically, you need to handle this error and return an appropriate error message

with current code, the Delete request won't fail because after decoding the pool id we first check the pool is present or not if it's not present we will return success.

Reply:
I have added the handling of ErrPoolNotFound in the calling function now. Is that what you mean?

Reply:
@Madhu-1 did I address your concern, or did I misunderstand?

Reply:
@nixpanic we also need to handle `GetObjectUUIDData`  https://github.com/ceph/ceph-csi/blob/master/pkg/util/voljournal.go#L434 . whoever is calling `GetObjectUUIDData` may also need to return proper error message also

Reply:
@nixpanic can you please address above ^^^^

Reply:
done!